Code/Source/VehicleDynamics/VehicleInputsState.h

```diff
@@ -7,12 +7,60 @@
  */
 #pragma once
 
+#include <AzCore/Time/ITime.h>
+#include <type_traits>
+//#include <concepts> //C++20
+
 namespace VehicleDynamics
 {
+    /* C++ 20
+    template<typename T>
+    concept Zeroable = std::integral<T> or std::floating_point<T>; // arithmetic types
+
+    template<typename T>
+    requires Zeroable<T> class InputZeroedOnTimeout
+    */
+
+    //! Inputs with an expiration date - effectively is zero after a certain time since update
+    template<typename T, typename = typename std::enable_if<std::is_arithmetic<T>::value, T>::type>
+    class InputZeroedOnTimeout
+    {
+    public:
+        InputZeroedOnTimeout(int64_t timeoutUs = 200000)
+            : m_timeoutUs(timeoutUs)
+        {
+        }
+
+        void UpdateValue(T updatedInput)
+        {
+            m_input = updatedInput;
+            m_lastUpdateUs = GetTimeSinceStartupUs();
+        }
+
+        T GetValue() const
+        {
+            if (auto timeSinceStartUs = GetTimeSinceStartupUs(); timeSinceStartUs - m_lastUpdateUs > m_timeoutUs)
+            {
+                return 0;
+            }
+            return m_input;
+        }
+
+    private:
+        int64_t GetTimeSinceStartupUs() const
+        {
+            return static_cast<int64_t>(AZ::Interface<AZ::ITime>::Get()->GetElapsedTimeUs());
+        }
+
+        int64_t m_timeoutUs;
```

Comment:
```suggestion
     const int64_t m_timeoutUs;
```
I also do not understand the need for SFINAE in this class.

Reply:
Primary reason for including it is that it informs the developer of the intent for the templated type without having him look into implementation. 
Also, compilation errors for invalid type are different, but I guess it is up to personal preference which is easier to understand.

Reply:
Another issue is that some non-arithmetic types can have zero assigned, which is not something we would like (e.g. std::string).